v2/controllers/addresspool_controller.go

```diff
@@ -51,6 +53,28 @@ func (r *AddressPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	logger.Info("synchronized")
+
+	if !ap.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
DeletionTimestamp should be compare to `nil`.

Reply:
and it is simpler to negate the logic:

```go
if ap.DeletionTimpstamp == nil {
    return ctrl.Result{}, nil
}

...
```

Reply:
Ah, this line is copied from the [kubebuilder book](https://book.kubebuilder.io/reference/using-finalizers.html).
Though the book uses `IsZero()`, it seems unnatural, so I'll fix it.

A failure in eviction may leave the zero value in the deletion timestamp, but I'm not sure this is meaningful.
https://github.com/kubernetes/kubernetes/issues/54525